src/btree/node.rs

```diff
@@ -505,81 +505,97 @@ impl Node {
 	}
 
 	pub fn seek(
-		from: Self,
+		mut self,
 		key: &[u8],
 		values: TablesRef,
 		log: &impl LogQuery,
-		depth: u32,
+		mut depth: u32,
 		stack: &mut Vec<(usize, NodeType, Self)>,
 		seek_to: SeekTo,
 	) -> Result<()> {
-		let (at, i) = from.position(key)?;
-		if at {
-			stack.push(match seek_to {
-				SeekTo::At => (i, NodeType::Separator, from),
-				SeekTo::After => (i + 1, NodeType::Child, from),
-			});
-			return Ok(())
-		}
-
-		if depth != 0 {
-			if let Some(child) = from.fetch_child(i, values, log)? {
-				stack.push((i, NodeType::Separator, from));
-				return Self::seek(child, key, values, log, depth - 1, stack, seek_to)
+		loop {
+			let (at, i) = self.position(key)?;
+			if at {
+				stack.push(match seek_to {
+					SeekTo::At => (i, NodeType::Separator, self),
+					SeekTo::After => (i + 1, NodeType::Child, self),
+				});
+				return Ok(())
+			}
+			if depth == 0 {
+				stack.push((i, NodeType::Separator, self));
+				return Ok(())
 			}
+
+			let child = if let Some(child) = self.fetch_child(i, values, log)? {
+				child
+			} else {
```

Comment:
```suggestion
			} else {
                               stack.push((i, NodeType::Separator, from));
```

Reply:
Actually when I look at it this line does not make much sense, I wonder if the original code doing it was not the origin of the stack overflows?

Reply:
hum not really, removing this comment.